scubagoggles/rego/Drive.rego

```diff
@@ -0,0 +1,1588 @@
+}
+
+tests contains {
+    "PolicyId": DriveId2_1,
+    "Criticality": "Should",
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs2_1, NonCompliantGroups2_1),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs2_1, "NonCompliantGroups": NonCompliantGroups2_1},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check2_1_OK
+    Conditions := {count(NonCompliantOUs2_1) == 0, count(NonCompliantGroups2_1) == 0 }
+    Status := (false in Conditions) == false
+}
+#--
+
+#
+# Baseline GWS.DRIVEDOCS.2.2
+#--
+
+DriveId2_2 := utils.PolicyIdWithSuffix("GWS.DRIVEDOCS.2.2")
+
+LogMessage2_2 := "Shared Drive Creation new_team_drive_restricts_cross_domain_access"
+
+Check2_2_OK if {
+    not PolicyApiInUse
+    events := utils.FilterEventsOU(LogEvents, LogMessage2_2, utils.TopLevelOU)
+    count(events) > 0
+}
+
+Check2_2_OK if {PolicyApiInUse}
+
+NonComplianceMessage2_2 := "Users outside the organization can access files in shared drives."
+
+NonCompliantOUs2_2 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage2_2
+}
+if {
+    not PolicyApiInUse
+    some OU in utils.OUsWithEvents
+    Events := utils.FilterEventsOU(LogEvents, LogMessage2_2, OU)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    contains("true", LastEvent.NewValue) == false
+    LastEvent.NewValue != "DELETE_APPLICATION_SETTING"
+}
+
+NonCompliantGroups2_2 contains {
+    "Name": Group,
+    "Value": NonComplianceMessage2_2
+}
+if {
+    not PolicyApiInUse
+    some Group in utils.GroupsWithEvents
+    Events := utils.FilterEventsGroup(LogEvents, LogMessage2_2, Group)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    contains("true", LastEvent.NewValue) == false
+    LastEvent.NewValue != "DELETE_APPLICATION_SETTING"
+}
+
+NonCompliantOUs2_2 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage2_2
+}
+if {
+    some OU, settings in input.policies
+    DriveEnabled(OU)
+    externalAccess := settings.drive_and_docs_shared_drive_creation.allowExternalUserAccess
+    externalAccess != false
+}
+
+tests contains {
+    "PolicyId": DriveId2_2,
+    "Criticality": "Should",
+    "ReportDetails": utils.NoSuchEventDetails(DefaultSafe, utils.TopLevelOU),
+    "ActualValue": "No relevant event for the top-level OU in the current logs",
+    "RequirementMet": DefaultSafe,
+    "NoSuchEvent": true
+}
+if {
+    not PolicyApiInUse
+    DefaultSafe := false
+    not Check2_2_OK
+}
+
+tests contains {
+    "PolicyId": DriveId2_2,
+    "Criticality": "Should",
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs2_2, NonCompliantGroups2_2),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs2_2, "NonCompliantGroups": NonCompliantGroups2_2},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check2_2_OK
+    Conditions := {count(NonCompliantOUs2_2) == 0, count(NonCompliantGroups2_2) == 0 }
+    Status := (false in Conditions) == false
+}
+#--
+
+#
+# Baseline GWS.DRIVEDOCS.2.3
+#--
```

Comment:
The report passes no matter what you select for "Allow people who aren't shared drive members to be added to files"

Reply:
![image](https://github.com/user-attachments/assets/440208ab-49b0-405e-830e-4d9df0ffeded)

Reply:
The org we use for testing uses Google Workspace Enterprise Plus, so that isn't the issue.

Reply:
After implementing Google's policy "reduction" process, this is working for me
when I change settings in the Admin UI.